WooCommerce/Classes/ViewRelated/Products/ProductDetailsViewController.swift

```diff
@@ -253,14 +265,66 @@ private extension ProductDetailsViewController {
             let size = CGSize(width: tableView.frame.width, height: Metrics.emptyProductImageHeight)
             mainImageView.image = StyleManager.wooWhite.image(size)
         }
+
+        if product.productStatus != .publish {
+            cell.textBadge?.applyPaddedLabelSubheadStyles()
+            cell.textBadge?.layer.backgroundColor = StyleManager.defaultTextColor.cgColor
+            cell.textBadge?.textColor = StyleManager.wooWhite
+            cell.textBadge?.text = product.productStatus.description
+        }
     }
 
     func configureProductName(cell: TitleBodyTableViewCell) {
         cell.accessoryType = .none
         cell.selectionStyle = .none
         cell.titleLabel?.text = NSLocalizedString("Title", comment: "Product details screen — product title descriptive label")
+        cell.bodyLabel?.applySecondaryBodyStyle()
         cell.bodyLabel?.text = product.name
     }
+
+    func configureTotalOrders(cell: TwoColumnTableViewCell) {
+        cell.selectionStyle = .none
+        cell.leftLabel?.text = NSLocalizedString("Total Orders", comment: "Product details screen - total orders descriptive label")
+        cell.rightLabel?.applySecondaryBodyStyle()
+        cell.rightLabel.textInsets = UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 0)
+        cell.rightLabel?.text = String(product.totalSales)
+    }
+
+    func configureReviews(cell: ProductReviewsTableViewCell) {
+        cell.selectionStyle = .none
+        cell.reviewLabel?.text = NSLocalizedString("Reviews", comment: "Reviews descriptive label")
+
+        cell.reviewTotalsLabel?.applySecondaryBodyStyle()
+        // 🖐🏼 I solemnly swear I'm not converting currency values to a Double.
+        let ratingCount = Double(product.ratingCount)
+        cell.reviewTotalsLabel?.text = ratingCount.humanReadableString()
+        let averageRating = Double(product.averageRating)
+        cell.starRatingView.rating = CGFloat(averageRating ?? 0)
+    }
+
+    func configurePermalink(cell: BasicTableViewCell) {
```

Comment:
In this method and the following, the cell is styled the same way, as far as I can see. Would it make sense to make it a "domain" cell (maybe a subclass of BasicTableViewCell)?  On the one hand that moght make the code here simpler and with richer semantics, on the other hand, it's an extra cell that I am not sure if we might use anywhere else... 🤷‍♂

Reply:
Yeah! Good idea. I can see this specific type of cell getting used a lot in the future.